optuna/trial.py

```diff
@@ -1253,11 +1260,18 @@ def suggest_discrete_uniform(self, name, low, high, q):
         discrete = distributions.DiscreteUniformDistribution(low=low, high=high, q=q)
         return self._suggest(name, discrete)
 
-    def suggest_int(self, name, low, high, step=1):
-        # type: (str, int, int, int) -> int
-        sample = self._suggest(
-            name, distributions.IntUniformDistribution(low=low, high=high, step=step)
-        )
+    def suggest_int(self, name, low, high, log=False, step=1):
```

Comment:
This is __NOT__ backward-compatible. We need to discuss the order with users and developers. I know this order aligns with `suggest_float` and makes sense BUT personally the backward compatibility is prioritized.
https://optuna.readthedocs.io/en/latest/reference/trial.html#optuna.trial.Trial.suggest_float

Reply:
How about introducing `log` after the `step` to keep the backward compatibility and showing the warning `log` and `step` will be keyword-only arguments? I think it makes users have time to migrate from positional arguments to keyword-only arguments with keeping the compatibility. (it's just my opinion)

Reply:
How about introducing `log` after the `step` to keep the backward compatibility and showing the warning `log` and `step` will be keyword-only arguments? I think it makes users have time to migrate from positional arguments to keyword-only arguments with keeping the compatibility. (it's just my opinion)

Reply:
Thank you both @crcrpar and @himkt for your comment. I agree with maintaining backward-compatibility. But I don't know how to do that. @himkt's suggestion sounds nice.

Reply:
Yeah, keyword only argument sounds good.

Reply:
I have changed them with keyword-only arguments.